test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -1632,6 +1633,7 @@ protected Settings nodeSettings(int nodeOrdinal) {
         Settings.Builder builder = Settings.builder()
                 // Default the watermarks to absurdly low to prevent the tests
                 // from failing on nodes without enough disk space
+                .put(NodeEnvironment.MAX_LOCAL_STORAGE_NODES_SETTING.getKey(), 8)
```

Comment:
I pushed fe563c5a940c5fc77fe59993b3f54253ae89fa2e and then on further thought pushed fdee065f854406d07d6fa01c56100b88fb56515f.